r/R/record-batch.R

```diff
@@ -189,3 +189,59 @@ record_batch <- RecordBatch$create
 
 #' @export
 names.RecordBatch <- function(x) x$names()
+
+#' @export
+rbind.RecordBatch <- function(...) {
+  abort("Use `Table$create()` to combine record batches")
+}
+
+cbind_check_length <- function(inputs, call = caller_env()) {
+  sizes <- map(inputs, function(input) {
+    # TODO: can we somehow implement vec_size for Arrow structures?
+    if (inherits(input, "ArrowTabular") || is.data.frame(input)) {
+      nrow(input)
+    } else {
+      length(input)
+    }
+  })
```

Comment:
This is why I suggested using `NROW()`, it does this for you: 

```
> NROW
function (x) 
if (length(d <- dim(x))) d[1L] else length(x)
<bytecode: 0x14dfdaf28>
<environment: namespace:base>
```

```suggestion
  sizes <- map_int(inputs, NROW)
```